pkg/pillar/types/volumetypes.go

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) 2020 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package types
+
+import (
+	"fmt"
+	"time"
+
+	zconfig "github.com/lf-edge/eve/api/go/config"
+	"github.com/lf-edge/eve/pkg/pillar/base"
+	uuid "github.com/satori/go.uuid"
+	log "github.com/sirupsen/logrus"
+)
+
+// VolumeConfig specifies the needed information for volumes
+type VolumeConfig struct {
+	VolumeID                uuid.UUID
+	ContentID               uuid.UUID
+	VolumeContentOriginType zconfig.VolumeContentOriginType
+	MaxVolSize              uint64
+	ReadOnly                bool
+	GenerationCounter       int64
+	DisplayName             string
+}
+
+// Key is volume UUID which will be unique
+func (config VolumeConfig) Key() string {
+	return fmt.Sprintf("%s.%d", config.VolumeID.String(), config.GenerationCounter)
+}
+
+// LogCreate :
+func (config VolumeConfig) LogCreate() {
+	logObject := base.NewLogObject(base.VolumeConfigLogType, config.DisplayName,
+		config.VolumeID, config.LogKey())
+	if logObject == nil {
+		return
+	}
+	logObject.CloneAndAddField("content-id", config.ContentID).
+		Infof("Volume config create")
+}
+
+// LogModify :
+func (config VolumeConfig) LogModify(old interface{}) {
+	logObject := base.EnsureLogObject(base.VolumeConfigLogType, config.DisplayName,
+		config.VolumeID, config.LogKey())
+
+	oldConfig, ok := old.(VolumeConfig)
+	if !ok {
+		log.Errorf("LogModify: Old object interface passed is not of VolumeConfig type")
+	}
+	if oldConfig.ContentID != config.ContentID ||
+		oldConfig.MaxVolSize != config.MaxVolSize ||
+		oldConfig.GenerationCounter != config.GenerationCounter {
+
+		logObject.CloneAndAddField("content-id", config.ContentID).
+			AddField("maxVolSize", config.MaxVolSize).
```

Comment:
These are new, right?
Two things 1. we don't use camelCase but dashes, and 2. integers should have a name ending in -int64
(so that the type can be determined from the name, otherwise it is treated as a string).

Reply:
>These are new, right?

Yes. Resolved.